src/workflows/registration.ts

```diff
@@ -58,10 +58,10 @@ export async function registerOffchainWorkflow({
 }
 
 export async function isRegisteredOnChainWorkflow(
-  starkPublicKey: string,
+  starkPublicKey: string | Promise<string>,
```

Comment:
what is this? if it is a promise, it should be function i think instead of a noun

Reply:
Yes, we should pass the string as a param because you can always await to get the promise resolved before calling the function.

Reply:
A better approach here is to await the l2Address in the function caller and just pass it as a string. 
My bad, folks.